lib/ssl/src/ssl_cipher.erl

```diff
@@ -321,45 +321,42 @@ block_decipher(Fun, #cipher_state{key=Key, iv=IV} = CipherState0,
 %%
 %% Description: Returns a list of supported cipher suites.
 %%--------------------------------------------------------------------
-suites({3, Minor}) ->
-    tls_v1:suites(Minor);
-suites({_, Minor}) ->
-    dtls_v1:suites(Minor).
-all_suites({3, 4} = Version) ->
-    suites(Version)
-	++ tls_v1:psk_suites({3,3})
-	++ tls_v1:srp_suites({3,3})
-        ++ tls_v1:rsa_suites({3,3})
-        ++ tls_v1:des_suites({3,3})
-        ++ tls_v1:rc4_suites({3,3});
-all_suites({3, _} = Version) ->
-    suites(Version)
-	++ tls_v1:psk_suites(Version)
-	++ tls_v1:srp_suites(Version)
-        ++ tls_v1:rsa_suites(Version)
-        ++ tls_v1:des_suites(Version)
-        ++ tls_v1:rc4_suites(Version);
+suites(?'TLS-1.X'=Version) ->
+    tls_v1:suites(Version);
+suites(?'DTLS-1.X'=Version) ->
+    dtls_v1:suites(Version).
+all_suites(?'TLS-1.3' = Version) ->
+    suites(Version) ++ tls_testing_suites(?'TLS-1.2');
+all_suites(?'TLS-1.X' = Version) ->
+    suites(Version) ++ tls_testing_suites(?'TLS-1.2');
```

Comment:
this is a bug! This should be `suites(Version) ++ tls_testing_suites(Version);`
phew!

Reply:
I think they should be called leagcy_suites rather than testing_suites